pkgs/games/mudlet/default.nix

```diff
@@ -1,46 +1,62 @@
-{ fetchurl, unzip, stdenv, makeWrapper, qtbase, yajl, libzip, hunspell
-, boost, lua5_1, luafilesystem, luazip, lrexlib-pcre, luasql-sqlite3, qmake }:
+{ fetchFromGitHub, fetchpatch, stdenv, wrapQtAppsHook, pcre, pugixml, qtbase, qtmultimedia, qttools, yajl, libzip, hunspell
+, boost, libGLU, lua, cmake,  which, }:
 
+let
+  luaEnv = lua.withPackages(ps: with ps; [ luazip luafilesystem lrexlib-pcre luasql-sqlite3 lua-yajl luautf8 ]);
+in
 stdenv.mkDerivation rec {
   pname = "mudlet";
-  version = "3.0.0-delta";
-
-  src = fetchurl {
-    url = "https://github.com/Mudlet/Mudlet/archive/Mudlet-${version}.tar.gz";
-    sha256 = "08fhqd323kgz5s17ac5z9dhkjxcmwvcmvhzy0x1vw4rayhijfrd7";
+  version = "4.0.3";
+
+  src = fetchFromGitHub {
+    owner = "Mudlet";
+    repo = "Mudlet";
+    rev = "Mudlet-${version}";
+    fetchSubmodules = true;
+    sha256 = "18bl4k0qgh47d9k5ipfvypfj1il678c0ws64a8adn8k21jajzkik";
   };
 
-  nativeBuildInputs = [ makeWrapper qmake ];
+  patches = [
+    ( fetchpatch {
+      url = "https://github.com/abbradar/Mudlet/commit/4f9450e9f66ac72f502e52951ff04b8ac8d123b3.patch";
```

Comment:
may be best for the patch to live in nixpkgs ?

Reply:
They already merged it upstream actually: https://github.com/Mudlet/Mudlet/commit/6306f99a847416dc8a690e57552553136ca77ea6

Reply:
I had it in nixpkgs but @abbradar wanted me to pull it from their repo.

I think i like it better this way because you could access more information about the patch when you know how to handle the link.

Reply:
the issue is that url points towards @abbradar url. so if he deletes his fork or the branch, the package will stop compiling. If the patch were in nixpkgs then we are in control. Referencing the upstreaming patch should be fine as if their commit disappear, it means we will have problems anyway.

Reply:
Ah, got it. fetching the patch from mudlet/mudlet now.

Reply:
Ah, got it. fetching the patch from mudlet/mudlet now.